### ISSUE TITLE
Str/Bytes incompatibility in single_project

### ISSUE THREAD

**Issue by mgrady3** [enthought/envisage#245]
In the `envisage.ui.single_project` code, the following line appears to cause issues with type mismatch:

https://github.com/enthought/envisage/blob/master/envisage/ui/single_project/project.py#L215

* `clean_filename()` returns bytes
* `get_default_path()` returns unicode

So these two can not be joined on [line 216](https://github.com/enthought/envisage/blob/master/envisage/ui/single_project/project.py#L216)

To reproduce the error - create a new python3 env and install the latest `envisage` (4.9.0), then do the following:

```
>>> import envisage
>>> envisage.__version__
'4.9.0'
>>> from envisage.ui.single_project.project import Project
>>> class MyProject(Project):
            pass
>>> prj = MyProject()
>>> loc = prj.get_default_project_location(None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/mgrady/.edm/envs/test-envisage/lib/python3.6/site-packages/envisage/ui/single_project/project.py", line 216, in get_default_project_location
    location = os.path.join(path, name)
  File "/Users/mgrady/.edm/envs/test-envisage/Resources/Python.app/Contents/MacOS/../../../../lib/python3.6/posixpath.py", line 94, in join
    genericpath._check_arg_types('join', a, *p)
  File "/Users/mgrady/.edm/envs/test-envisage/Resources/Python.app/Contents/MacOS/../../../../lib/python3.6/genericpath.py", line 151, in _check_arg_types
    raise TypeError("Can't mix strings and bytes in path components") from None
TypeError: Can't mix strings and bytes in path components
```

PR to follow shortly resolving this issue.

**Comment by mgrady3** [comment]
Leaving this open for the time being. While the PR in `Envisage` has been closed, a fix PR has been introduced in the `Traits` codebase - see https://github.com/enthought/traits/pull/589

**Comment by mdickinson** [comment]
Let's fix this code so that it doesn't import the utility from Traits at all.

**Comment by mgrady3** [comment]
I'd be happy to make a PR for that this afternoon or this weekend.

**Comment by mdickinson** [comment]
Closed by #252.
